### PR TITLE
SPARK-38661: [TESTS][FOLLOWUP] Replace Symbol usage with $"" in unit tests

### PR DESCRIPTION
**Pull request by martin-g**
### What changes were proposed in this pull request?

Replace symbols like 'abc and Symbol("abc") (where possible) with the more verbose $"abc" in the test code.

### Why are the changes needed?

This is a follow-up of #35560

Building with Scala 2.13 produces a lot of warnings like the following ones:

```
[warn] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala:562:11: [deprecation @  | origin= | version=2.13.0] symbol literal is deprecated; use Symbol("d") instead
[warn]           'd.cast("string"),
[warn]           ^
[warn] /home/runner/work/spark/spark/sql/core/src/test/scala/org/apache/spark/sql/execution/BaseScriptTransformationSuite.scala:563:11: [deprecation @  | origin= | version=2.13.0] symbol literal is deprecated; use Symbol("e") instead
[warn]           'e.cast("string")).collect())
```

This should make it easier to upgrade to Scala 3 later.

### Does this PR introduce _any_ user-facing change?

No! The PR touches only test classes!

### How was this patch tested?

The build at CI must be green!

### CODE REVIEW

**Comment by srowen** [comment]
Thanks @martin-g ! this is larger than the original change, so is it safe to assume that this implements `'foo` -> `$"foo"` basically everywhere in tests?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala [hunk 1]
```diff
@@ -148,7 +148,7 @@ class CatalystTypeConvertersSuite extends SparkFunSuite with SQLHelper {
   }
 
   test("SPARK-24571: convert Char to String") {
-    val chr: Char = 'X'
+    val chr: Char = $"X"'
```
Comment (awdavidson) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala:151]: Doesn’t look like this should have been changed?

**Comment by martin-g** [comment]
> basically everywhere in tests?

This PR also changes `Symbol("foo")` to `$"foo", where possible.

There are more tests which still use `'foo`. Should I add them to this PR or leave them for another follow-up ?

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala (hunk 1 above)
Comment (srowen) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala:151]: @martin-g could you have a look at these? does seem like the search and replace failed in a few cases

Comment (martin-g) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/CatalystTypeConvertersSuite.scala:151]: Yes! It is possible that there are such regressions!
Please don't spend time on reviewing the PR until I change it from `Draft` to `Ready for review` !

**Review by srowen — CHANGES_REQUESTED** [review]
I did skim all of it (wow, big change) and there are maybe a few things we need to fix, but mostly looking good

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/EncoderResolutionSuite.scala [hunk 2]
```diff
@@ -200,28 +200,29 @@ class EncoderResolutionSuite extends PlanTest {
...
     encoder.resolveAndBind(attrs)
   }
 
   test("SPARK-28497: complex type is not compatible with string encoder schema") {
     val encoder = ExpressionEncoder[String]
 
-    Seq('a.struct('x.long), 'a.array(StringType), 'a.map(StringType, StringType)).foreach { attr =>
-      val attrs = Seq(attr)
-      assert(intercept[AnalysisException](encoder.resolveAndBind(attrs)).message ==
+    Seq($"a".struct($"x".long), $"a".array(StringType), Symbol("a").map(StringType, StringType))
```
[the remark this answers is not in the document]
Comment (martin-g) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/encoders/EncoderResolutionSuite.scala:210]: It seems some implicit is missing. I had to revert from `$"a".map(...)` to `Symbol("a").map(...)` to be able to compile.
If you/someone knows the missing implicit please let me know and I will rework it!

**Comment by martin-g** [comment]
@srowen @HeartSaVioR @awdavidson I think the PR is ready for review now!

**Comment by awdavidson** [comment]
I echo @srowen pretty big change but generally looking ok. Let’s address test failure

Not to increase the scope of this change but a follow up could be to add a scalastyle check warning against the use of `‘foo` declaration mainly to avoid it being reintroduced going forward?

**Comment by martin-g** [comment]
> Do you feel pretty good that there aren't other typos?

I've checked all changes and they look OK to me !

The last test failure was because `$"foo bar"` is not supported. It should stay `Symbol("foo bar")`.

**Review by srowen — APPROVED** [review]

**Comment by srowen** [comment]
Merged to master

**Comment by martin-g** [comment]
Thank you for your help, @srowen !

**Pull request closed without merging**